From 39bd599003f8e8b39dc7dd6db0540399f91a3ddc Mon Sep 17 00:00:00 2001
From: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
Date: Wed, 11 Feb 2015 03:19:11 +0300
Subject: [PATCH 32/44] ARM: sa1100: add dummy clk_set_rate implementation

Add dummy clk_set_rate implementation that returns -EINVAL if the rate
passed is not equal to the rate of the clock. This for example fixes the
following build issue:

ERROR: "clk_set_rate" [drivers/usb/phy/phy-generic.ko] undefined!
scripts/Makefile.modpost:90: recipe for target '__modpost' failed

Cc: <stable@vger.kernel.org>
Signed-off-by: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
---
 arch/arm/mach-sa1100/clock.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-sa1100/clock.c b/arch/arm/mach-sa1100/clock.c
index cbf53bb..dcffa9a 100644
--- a/arch/arm/mach-sa1100/clock.c
+++ b/arch/arm/mach-sa1100/clock.c
@@ -104,6 +104,15 @@ unsigned long clk_get_rate(struct clk *clk)
 }
 EXPORT_SYMBOL(clk_get_rate);
 
+int clk_set_rate(struct clk *clk, unsigned long rate)
+{
+	if (rate != clk_get_rate(clk))
+		return -EINVAL;
+
+	return 0;
+}
+EXPORT_SYMBOL(clk_set_rate);
+
 const struct clkops clk_gpio27_ops = {
 	.enable		= clk_gpio27_enable,
 	.disable	= clk_gpio27_disable,
-- 
1.9.1

